telemetry/telemetry.go

```diff
@@ -27,10 +28,11 @@ var (
 	// Client key; set at link-time with -X flag
 	defaultAmplitudeKey = ""
 
-	// Store the distinct ID; run through the process
+	// Store the user ID and team ID; run through the process
 	// of getting it only once.
-	userDistinctID   string
-	userDistinctOnce sync.Once
+	userID           string
+	teamID           string
+	userIdentityOnce sync.Once
```

Comment:
I can't find this sync.Once used anywhere in this PR.

Do you believe it is no longer necessary (because Init is only invoked once)?  I do not have a high degree of confidence that telemetry.Blah() will never be called before Init() and would like to have something sensible happen. Using the nullClient{} is probably fine?

Probably we can just delete this, but please do think about synchronization issues.  The Go memory model does allow us to write to analyticsClient (and userID, and teamID) before starting any other goroutines that use them.  Is this sufficient (as I evidently believed in a previous version) or do we need to do explicit locking?

Reply:
* I have reused `sync.Once` to ensure the client is initiated before sending the events.
  * There is one minor issue related to the init process logging. If tomorrow we add a new `telemetry.Init(false)` call and somehow `telemetry.tryTrackingEvent()` is called before this, we will show logs.
* Does this seem okay, or should we explore the `chan`? This could lock the `tryTrackingEvent()` function's routine